spectrum-offchain-lm/src/executor.rs

```diff
@@ -184,19 +184,33 @@ where
                         match self.prover.sign(tx) {
                             Ok(tx) => {
                                 trace!(target: "offchain_lm", "Transaction ID for order [{}] is [{}]", ord.get_self_ref(), tx.id());
-                                if let Err(client_err) = self.network.submit_tx(tx).await {
-                                    // Note, here `submit_tx(tx)` can fail not only bc pool state is consumed,
-                                    // but also bc bundle is consumed, what is less possible though. That's why
-                                    // we just invalidate pool.
-                                    // todo: In the future more precise error handling may be possible if we
-                                    // todo: implement a way to find out which input failed exactly.
+                                if let Err(client_err) = self.network.submit_tx(tx.clone()).await {
                                     warn!("Execution failed while submitting tx due to {}", client_err);
-                                    self.pool_repo
-                                        .lock()
-                                        .await
-                                        .invalidate(pool.get_self_state_ref(), pool.get_self_ref())
-                                        .await;
-                                    self.backlog.lock().await.recharge(ord).await;
+                                    if let Some(missing_indices) = parse_err(&client_err.0) {
+                                        for idx in missing_indices {
+                                            // Staking bundles start at index 2 of inputs.
+                                            if idx > 1 {
```

Comment:
What if pool and bundle inputs are missing at the same time? I think we should make some kind of abstraction that would allow to describe simple decision trees for each kind of order (e.g. if inputs[0] is missing we should always invalidate pool, if input[1] is missing we should invalidate order in case of Deposit|Redeem, in the case of Compound we should invalidate funding box etc...). And then use it to handle missing inputs here. What I see here looks more like an ad-hoc solution for a special case.